core/src/main/java/hudson/util/ProcessTree.java

```diff
@@ -171,6 +176,28 @@ public void killAll(Process proc, Map<String, String> modelEnvVars) throws Inter
             }
         return killers;
     }
+    
+    final List<ProcessKillingVeto> getVetoers() throws InterruptedException {
+        if (vetoers==null)
+            try {
+                VirtualChannel channelToMaster = SlaveComputer.getChannelToMaster();
+                if (channelToMaster!=null) {
+                    vetoers = channelToMaster.call(new SlaveToMasterCallable<List<ProcessKillingVeto>, IOException>() {
+                        public List<ProcessKillingVeto> call() throws IOException {
+                            return new ArrayList<ProcessKillingVeto>(ProcessKillingVeto.all());
+                        }
+                    });
+                } else {
+                    // used in an environment that doesn't support talk-back to the master.
+                    // let's do with what we have.
+                    vetoers = Collections.emptyList();
+                }
+            } catch (IOException e) {
+                LOGGER.log(Level.WARNING, "Failed to obtain vetoers",e);
+                vetoers = Collections.emptyList();
```

Comment:
It would be a fatal error since Jenkins won't retry to collect the list within the same class instance